### PR TITLE
Support Functions for customType option

### PR DESCRIPTION
**Pull request by Uzlopak**
Well. Why not?

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,14 +99,24 @@ await fastify.register(
...
   { customTypes: /x-protobuf$/ }
 )
 ```
 
+or
+
+```javascript
+await fastify.register(
+  import('@fastify/compress'),
+  { customTypes: value => value.endsWith('x-protobuf') }
```
Comment (Eomm) [inline README.md:116]:
```suggestion
  { customTypes: contentType => contentType.endsWith('x-protobuf') }
```

File: types/index.d.ts [hunk 2]
```diff
@@ -59,11 +59,13 @@ type RouteDecompressOptions = Pick<fastifyCompress.FastifyCompressOptions,
 
 type EncodingToken = 'br' | 'deflate' | 'gzip' | 'identity';
 
+type CompressibleContentTypeFunction = (value: string) => boolean;
```
Comment (Eomm) [inline types/index.d.ts:62]:
```suggestion
type CompressibleContentTypeFunction = (contentType: string) => boolean;
```

**Review by Eomm — COMMENTED** [review]
just suggested a clearer var name 👍🏼

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Uzlopak** [comment]
@Eomm 
PTAL

**Review by Eomm — APPROVED** [review]

**Pull request merged**